### PR TITLE
Adding CloudBigtableService to abstract CBT services

### PR DESCRIPTION
**Pull request by sduskis**
I plan to move as much functionality out of the bigtable-hbase-dataflow
as I can.  Eventually, this functionality should be housed in a
different jar, and should be reusable for both dataflow and beam.

### CODE REVIEW

File: bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java [hunk 1]
```diff
@@ -27,6 +27,7 @@
 import com.google.bigtable.repackaged.com.google.com.google.bigtable.v2.RowRange;
 import com.google.bigtable.repackaged.com.google.com.google.bigtable.v2.RowSet;
 import com.google.bigtable.repackaged.com.google.protobuf.ByteString;
+import com.google.cloud.bigtable.batch.util.ByteStringUtil;
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableScanConfiguration.java]: typo?

File: bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/batch/util/CloudBigtableServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import java.io.IOException;
+import java.util.List;
+
+public class CloudBigtableServiceImpl implements CloudBigtableService {
+
+  @Override
+  public List<SampleRowKeysResponse> getSampleRowKeys(CloudBigtableTableConfiguration config)
+      throws IOException {
+    BigtableOptions bigtableOptions = config.toBigtableOptions();
+    try (BigtableSession session = new BigtableSession(bigtableOptions)) {
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/batch/util/CloudBigtableServiceImpl.java]: Is it possible to instantiate the session in the constructor?

File: bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/batch/util/package-info.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * <p>
+ * Google Cloud Bigtable offers you a fast, fully managed, massively scalable NoSQL database service
+ * that's ideal for web, mobile, and Internet of Things applications requiring terabytes to
+ * petabytes of data. Unlike comparable market offerings, Cloud Bigtable doesn't require you to
+ * sacrifice speed, scale, or cost efficiency when your applications grow. Cloud Bigtable has been
+ * battle-tested at Google for more than 10 years--it's the database driving major applications such
+ * as Google Analytics and Gmail.
+ * </p>
+ */
+package com.google.cloud.bigtable.batch.util;
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/batch/util/package-info.java]: batch.base or common, might be a better name.

File: bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/batch/util/CloudBigtableServiceImpl.java (hunk 2 above)
Comment (sduskis) [inline bigtable-dataflow-parent/bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/batch/util/CloudBigtableServiceImpl.java]: It is possible, but it's not necessary.  The BigtableSession has a shared channel pool.

**Pull request merged**